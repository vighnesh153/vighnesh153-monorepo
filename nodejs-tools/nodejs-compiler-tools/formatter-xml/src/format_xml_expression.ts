import {
  XmlCommentNode,
  XmlExpression,
  XmlPrologNode,
  XmlTagNode,
  XmlTextNode,
} from "@vighnesh153/parser-xml";
import { formatXmlPrologNode } from "./format_xml_prolog_node.ts";
import { formatXmlTagNode } from "./format_xml_tag_node.ts";
import { formatCommentNode } from "./format_comment_node.ts";
import { formatTextNode } from "./format_text_node.ts";

type FormatXmlExpressionProps = {
  readonly expression: Readonly<XmlExpression>;
  readonly indentation: number;
  readonly indentationLevel: number;
  readonly sortAttributes: boolean;
};

export function formatXmlExpression({
  expression,
  indentation,
  indentationLevel,
  sortAttributes,
}: FormatXmlExpressionProps): string {
  if (expression instanceof XmlPrologNode) {
    return formatXmlPrologNode({
      xmlPrologNode: expression,
      indentationLevel,
      indentation,
    });
  }
  if (expression instanceof XmlTagNode) {
    return formatXmlTagNode({
      xmlTagNode: expression,
      indentationLevel,
      indentation,
      sortAttributes,
    });
  }
  if (expression instanceof XmlCommentNode) {
    return formatCommentNode({
      commentNode: expression,
      indentationLevel,
      indentation,
    });
  }
  if (expression instanceof XmlTextNode) {
    return formatTextNode({
      textNode: expression,
      indentationLevel,
      indentation,
    });
  }
  throw new Error(`Unexpected xml expression: ${expression.toString(0)}`, {
    cause: expression,
  });
}
