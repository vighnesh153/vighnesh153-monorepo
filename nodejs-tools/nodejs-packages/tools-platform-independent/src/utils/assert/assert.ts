import { not } from '../not';

export function assert(condition: boolean, message = 'Assertion failed') {
  if (not(condition)) {
    throw new Error(message);
  }
}
