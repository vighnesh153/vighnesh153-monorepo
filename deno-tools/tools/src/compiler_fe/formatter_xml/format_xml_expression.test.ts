import { assertEquals } from "@std/assert";
import { assertSnapshot } from "@std/testing/snapshot";
import type {
  XmlCommentNode,
  XmlPrologNode,
  XmlTagNode,
} from "@/compiler_fe/parser_xml/mod.ts";
import { formatXmlExpression } from "./format_xml_expression.ts";
import { parseProgram } from "./test_utils.ts";

Deno.test("formatXmlExpression should format xml prolog node", async (t) => {
  const [parser, program] = parseProgram(
    `<? xml  encoding =   "utf-8"  version =  "2.0"   ?>`,
  );

  assertEquals(parser.errors.length, 0);
  assertEquals(program.statements.length, 1);
  await assertSnapshot(
    t,
    formatXmlExpression({
      expression: program.statements[0] as XmlPrologNode,
      indentation: 4,
      indentationLevel: 0,
      sortAttributes: true,
    }),
  );
});

Deno.test("formatXmlExpression should format xml tag node", async (t) => {
  const [parser, program] = parseProgram(
    `<manifest   simpleProperty = "some random value" deeply :  nested:property = "20" >
  </ manifest>
    `,
  );

  assertEquals(parser.errors.length, 0);
  assertEquals(program.statements.length, 1);
  await assertSnapshot(
    t,
    formatXmlExpression({
      expression: program.statements[0] as XmlTagNode,
      indentation: 4,
      indentationLevel: 0,
      sortAttributes: true,
    }),
  );
});

Deno.test("formatXmlExpression should format comment node", async (t) => {
  const [parser, program] = parseProgram(
    `<!-- <uses-permission android:name="com.android.providers.tv.permission.WRITE_EPG_DATA" /> -->`,
  );

  assertEquals(parser.errors.length, 0);
  assertEquals(program.statements.length, 1);
  await assertSnapshot(
    t,
    formatXmlExpression({
      expression: program.statements[0] as XmlCommentNode,
      indentation: 4,
      indentationLevel: 0,
      sortAttributes: true,
    }),
  );
});

Deno.test("formatXmlExpression should format text node", async (t) => {
  const [parser, program] = parseProgram(
    `< pokemon  >  Pikachu    < /  pokemon >`,
  );

  assertEquals(parser.errors.length, 0);
  assertEquals(program.statements.length, 1);
  await assertSnapshot(
    t,
    formatXmlExpression({
      expression: (program.statements[0] as XmlTagNode).children[0],
      indentation: 4,
      indentationLevel: 0,
      sortAttributes: true,
    }),
  );
});
