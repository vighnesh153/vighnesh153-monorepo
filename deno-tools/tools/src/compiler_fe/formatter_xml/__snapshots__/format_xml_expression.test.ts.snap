export const snapshot = {};

snapshot[`formatXmlExpression should format xml prolog node 1`] = `'<?xml encoding="utf-8" version="2.0"?>'`;

snapshot[`formatXmlExpression should format xml tag node 1`] = `
'<manifest
    simpleProperty="some random value"
    deeply:nested:property="20" />'
`;

snapshot[`formatXmlExpression should format comment node 1`] = `'<!-- <uses-permission android:name="com.android.providers.tv.permission.WRITE_EPG_DATA" /> -->'`;

snapshot[`formatXmlExpression should format text node 1`] = `"Pikachu"`;
